app/helpers/sessions_helper.rb

```diff
@@ -4,17 +4,46 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def handle_login user
+    log_in user
+    params[:session][:remember_me] == "1" ? remember(user) : forget(user)
+    redirect_to user
+  end
+
+  # Remembers a user in a persistent session.
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   # Returns the current logged-in user (if any).
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user&.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
   end
 
   def logged_in?
     !current_user.nil?
   end
 
+  # Forgets a persistent session.
+  def forget user
+    user.forget
+    cookies.delete(:user_id)
```

Comment:
```suggestion
    cookies.delete :user_id
```
chỗ nào bỏ dc ngoặc thì cứ bỏ e nhé
 k khuyến khích dùng ngoặc